### PR TITLE
Team Teaching - LOD shouldRender in render engine

### PR DESCRIPTION
**Pull request by Atlante45**

### CODE REVIEW

File: libraries/render-utils/src/Model.cpp [hunk 1]
```diff
@@ -310,10 +310,14 @@ void Model::init() {
...
+        
         _renderPipelineLib.addRenderPipeline(
-            RenderKey(RenderKey::IS_TRANSLUCENT),
-            modelVertex, modelTranslucentPixel);
+             RenderKey(RenderKey::IS_TRANSLUCENT),
+             modelVertex, modelTranslucentPixel);
+        // FIXME Ignore lightmap for translucents meshpart
+        _renderPipelineLib.addRenderPipeline(
+             RenderKey(RenderKey::IS_TRANSLUCENT | RenderKey::HAS_LIGHTMAP),
+             modelVertex, modelTranslucentPixel);
```
Comment (Atlante45) [inline libraries/render-utils/src/Model.cpp]: @samcake This is my hack around an unknown pipeline in the render engine. Thoughts?

File: libraries/entities/src/EntityItem.cpp [hunk 2]
```diff
@@ -662,7 +662,7 @@ void EntityItem::adjustEditPacketForClockSkew(unsigned char* editPacketBuffer, s
     memcpy(&lastEditedInLocalTime, dataAt, sizeof(lastEditedInLocalTime));
     quint64 lastEditedInServerTime = lastEditedInLocalTime + clockSkew;
     memcpy(dataAt, &lastEditedInServerTime, sizeof(lastEditedInServerTime));
-    #if 1 //def WANT_DEBUG
+    #ifdef WANT_DEBUG
```
Comment (Atlante45) [inline libraries/entities/src/EntityItem.cpp]: @ZappoMan Is it ok to turn that off?

**Comment by hifi-gustavo** [comment]
Build console output for commit: d431a1b522c865e69edfb4f6a8ee543cd0f453e3 available at: https://gist.github.com/74a89719800e82eea52a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5021.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5021.dmg

**Comment by samcake** [comment]
Perf are bad still on windows, maybe it's just my machine but i think there is something that just bork everything and it happened maybe yesterday or friday
i think it's good to go anyway beacause i see good perf almost on par with the main branch on mac

File: libraries/entities/src/EntityItem.cpp (hunk 2 above)
Comment (ZappoMan) [inline libraries/entities/src/EntityItem.cpp]: oops - thanks

File: interface/src/LODManager.cpp [hunk 3]
```diff
@@ -217,6 +217,47 @@ QString LODManager::getLODFeedbackText() {
...
+bool LODManager::shouldRender(const RenderArgs* args, const AABox& bounds) {
+    const float maxScale = (float)TREE_SCALE;
+    const float octreeToMeshRatio = 4.0f; // must be this many times closer to a mesh than a voxel to see it.
+    float octreeSizeScale = args->_sizeScale;
+    int boundaryLevelAdjust = args->_boundaryLevelAdjust;
+    float visibleDistanceAtMaxScale = boundaryDistanceForRenderLevel(boundaryLevelAdjust, octreeSizeScale) / octreeToMeshRatio;
+    float distanceToCamera = glm::length(bounds.calcCenter() - args->_viewFrustum->getPosition());
+    float largestDimension = bounds.getLargestDimension();
+    
+    static bool shouldRenderTableNeedsBuilding = true;
```
Comment (ZappoMan) [inline interface/src/LODManager.cpp]: why not use LODManager::shouldRenderMesh(float largestDimension, float distanceToCamera) here?? Looks like this code is all duplicated?

**Comment by ZappoMan** [comment]
I'm going to merge this as is... but I'd like us to revisit the LODManager::shouldRender() and LODManager::shouldRenderMesh() to see if we can consolidate them, since they appear to be the same or nearly identical

**Pull request merged**